lib/pbench/server/api/resources/query_apis/search_api.py

```diff
@@ -0,0 +1,126 @@
+            ),
+        )
+
+    def assemble(self, json_data: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
+        """
+        """
+        user = json_data.get("user")
+        start = json_data.get("start")
+        end = json_data.get("end")
+        query = json_data.get("query")
+        selected_fields = json_data("fields")
+
+        # We need to pass string dates as part of the Elasticsearch query; we
+        # use the unconverted strings passed by the caller rather than the
+        # adjusted and normalized datetime objects for this.
+        start_arg = f"{start:%Y-%m}"
+        end_arg = f"{end:%Y-%m}"
+
+        self.logger.info(
+            "Search query for user {}, prefix {}: ({} - {}) on query: {}",
+            user,
+            self.prefix,
+            start,
+            end,
+            query,
+        )
+
+        uri_fragment = self._gen_month_range("run", start, end)
+        return {
+            "path": f"/{uri_fragment}/_search",
+            "kwargs": {
+                "json": {
+                    "query": {
+                        "bool": {
+                            "filter": [
+                                {"term": self._get_user_term(user)},
+                                {
+                                    "range": {
+                                        "@timestamp": {"gte": start_arg, "lte": end_arg}
+                                    }
+                                },
+                            ],
+                            "must": {
+                                "query_string": {
+                                    "query": f"*{query}*",
+                                    "analyze_wildcard": "true",
+                                }
+                            },
+                        }
+                    },
+                    "sort": [
+                        {"@timestamp": {"order": "desc", "unmapped_type": "boolean"}}
+                    ],
+                    "_source": {"include": selected_fields},
+                },
+                "params": {"ignore_unavailable": "true"},
+            },
+        }
+
+    def postprocess(self, es_json: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
+        """
+        """
+        results = []
+        # If there are no matches for the user, query, and time range,
+        # return the empty list rather than failing.
+        try:
+            count = es_json["hits"]["total"]["value"]
+            if int(count) == 0:
+                self.logger.warning("No data returned by Elasticsearch")
+                return jsonify(result)
+        except KeyError as e:
+            raise PostprocessError(
+                f"Can't find Elasticsearch match data {e} in {es_json!r}"
+            )
+        except ValueError as e:
+            raise PostprocessError(f"Elasticsearch hit count {count!r} value: {e}")
+        hits = es_json["hits"]["hits"]
+        for hit in hits:
```

Comment:
I don't really see the value in the local `hits` here, since you only reference it once and `for hit in es_json["hits"]["hits"]` isn't any hard to read.